### PR TITLE
Master relational model pro

### PR DESCRIPTION
**Pull request by pro-odoo**

### CODE REVIEW

File: addons/web/static/src/views/list/list_controller.js [hunk 1]
```diff
@@ -107,11 +107,8 @@ export class ListController extends Component {
             rootRef: this.rootRef,
             beforeLeave: async () => {
                 const list = this.model.root;
-                const editedRecord = list.editedRecord;
-                if (editedRecord) {
-                    if (!(await list.unselectRecord(true))) {
-                        return false;
-                    }
+                if (!(await list.leaveEditMode())) {
+                    return false;
```
Comment (aab-odoo) [inline addons/web/static/src/views/list/list_controller.js]:
```suggestion
                return list.leaveEditMode();
```

**Review by aab-odoo — APPROVED** [review]

File: addons/web/static/src/views/list/list_controller.js (hunk 1 above)
Comment (pro-odoo) [reply addons/web/static/src/views/list/list_controller.js]: We do not need to wait for the saving result ?

Comment (aab-odoo) [reply addons/web/static/src/views/list/list_controller.js]: it's equivalent, in both cases you return a promise that resolves in a boolean (for the `false` case at least)

Comment (pro-odoo) [reply addons/web/static/src/views/list/list_controller.js]: Right, updated :)

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**